backend/app/api/src/helpers/SetupStepsUpdater.ts

```diff
@@ -0,0 +1,194 @@
+        platform: Platform) => void> = {
+        [SetupStepType.Groups]: this.updateStepGroups,
+        [SetupStepType.Premises]: this.updateStepPremises
+    }
+
+    static async updateSetupStepsForAllOrganizationsInCurrentPeriod({
+        batchSize, stepTypes, platforms
+    }: { batchSize?: number, stepTypes?: SetupStepType[], platforms?: Platform[] } = {}) {
+
+        if(!platforms) {
+            platforms = await Platform.all();
+        }
+
+        for (const platform of platforms) {
+            const periodId = platform.periodId;
+
+            let lastId = "";
+
+            // eslint-disable-next-line no-constant-condition
+            while (true) {
+                const organizationRegistrationPeriods =
+                    await OrganizationRegistrationPeriod.where(
+                        {
+                            id: { sign: ">", value: lastId },
+                            periodId: periodId,
+                        },
+                        { limit: batchSize ?? 10, sort: ["id"] }
+                    );
+
+                if (organizationRegistrationPeriods.length === 0) {
+                    lastId = "";
+                    break;
+                }
+
+                for (const organizationRegistrationPeriod of organizationRegistrationPeriods) {
+                    const organizationId =
+                        organizationRegistrationPeriod.organizationId;
+                    if (!organizationId) {
+                        continue;
+                    }
+                    console.log(
+                        "[FLAG-MOMENT] checking flag moments for " +
+                            organizationId
+                    );
+                    const organization = await Organization.getByID(
+                        organizationId
+                    );
+
+                    if (!organization) {
+                        continue;
+                    }
+
+                    await SetupStepUpdater.updateFor(
+                        organizationRegistrationPeriod,
+                        platform,
+                        organization,
+                        stepTypes
+                    );
+                }
+
+                lastId =
+                    organizationRegistrationPeriods[
+                        organizationRegistrationPeriods.length - 1
+                    ].id;
+            }
+        }
+    }
+
+    static async updateForOrganization(
+        organization: Organization,
+        {
+            platform,
+            organizationRegistrationPeriod,
+            stepTypes
+        }: {
+            platform?: Platform;
+            organizationRegistrationPeriod?: OrganizationRegistrationPeriod;
+            stepTypes?: SetupStepType[]
+        }
+    ) {
+        if (!platform) {
+            platform = (await Platform.all())[0];
+            if (!platform) {
+                console.error("No platform not found");
+                return;
```

Comment:
Gezien de platform op heel veel plaatsen nodig is, is er een cache om queries te voorkomen: `getSharedPrivateStruct` dit geeft wel een struct terug, maar normaal alle data die je nodig hebt